receiver/redisreceiver/metadata.yaml

```diff
@@ -5,8 +5,55 @@ attributes:
     description: Redis CPU usage state
   db:
     description: Redis database identifier
+  version:
+    description: Redis binary version
+  role:
+    description: Redis node's role, replica or primary
+  cmd:
+    description: Redis command name
 
 metrics:
+  redis.maxmemory:
+    enabled: true
+    description: The value of the maxmemory configuration directive
+    unit: By
+    gauge:
+      value_type: int
+
+  redis.version:
+    enabled: true
+    description: Redis binary version, metric value is fixed to be 1
+    gauge:
+      value_type: int
+    attributes: [version]
```

Comment:
I don't think this is the right way to report a redis version. Setting this the value as a resource attribute should be more appropriate.

Reply:
Adding version as a resource attribute will be a breaking change (given that we mark all the metrics as optional). Can you make that as a separate PR please?

Reply:
Do you mean remove version in this PR and do it in a separate PR as resource attribute? Or make it the way it currently be and change it to resource attribute in another PR?

BTW how would resource attribute finally be represented in an emitted metric? I didn't find a clear definition of it searching around. If I do want a version metric, what should I define? Could it be an up down counter as role?

Reply:
> Do you mean remove version in this PR and do it in a separate PR as resource attribute? Or make it the way it currently be and change it to resource attribute in another PR?

Let's remove the version metric from here and move it to a separate PR as a resource attribute.

> BTW how would resource attribute finally be represented in an emitted metric? I didn't find a clear definition of it searching around. If I do want a version metric, what should I define? Could it be an up down counter as role?

It'll be an attribute set on the resource (set of metrics). See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/zookeeperreceiver/metadata.yaml#L7-L9 as an example